app/models/activity.py

```diff
@@ -234,4 +236,7 @@ def retrieve_all_versions(self, max_reception_time=None):
 @event.listens_for(Activity, "after_insert")
 @event.listens_for(Activity, "after_update")
 def set_last_submitter_id(mapper, connect, target):
-    target.last_submitter_id = current_user.id
+    try:
+        target.last_submitter_id = current_user.id
+    except Exception as e:
+        target.last_submitter_id = target.submitter_id
```

Comment:
question: on ne peut pas le détecter autrement que par l'erreur ? peut être avec `current_user is not None` si c'est bien ça qui peut poser problème

Reply:
en fait c'est relou mais `current_user is not None` renvoie False même quand `current_user` résoud None... C'est lié au fait que c'est un proxy object. c'est pour ça que j'ai fait ça comme ça mais je suis d'accord que c'est pas ouf

Reply:
si tu n'as pas trouvé d'autres moyens, laissons comme ça :+1: